mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/DeleteMessageListener.java

```diff
@@ -164,7 +171,10 @@ public boolean isHandling(Event event) {
                     .map(ComposedMessageIdWithMetaData::getComposedMessageId)
                     .concatMap(metadata -> handleMessageDeletionAsPartOfMailboxDeletion((CassandraMessageId) metadata.getMessageId(), mailboxId)
                         .then(imapUidDAO.delete((CassandraMessageId) metadata.getMessageId(), mailboxId))
-                        .then(messageIdDAO.delete(mailboxId, metadata.getUid()))),
+                        .then(messageIdDAO.delete(mailboxId, metadata.getUid()))
+                        .then(threadLookupDAO.selectOneRow(metadata.getMessageId())
+                            .map(pair -> threadDAO.deleteSome(pair.getLeft(), pair.getRight())))
+                        .then(threadLookupDAO.deleteOneRow(metadata.getMessageId()))),
```

Comment:
Please write tests for this clean up:

 - You perform it only for mailbox deletion
 - If you perform the deletion after the message is deleted in case of failure, you will  never do the cleanup...

Reply:
CassandraMailboxManagerTest.DeletionTests